File: prob_2.php
<?php

/*
 * Wes Kilgour
 *
 * Problem 2
 * Even Fibonacci numbers
 *
 * https://projecteuler.net/problem=2
 */

$term1 = 0;
$term2 = 1;
$term3 = 1;

$evenSum = 0;

while ($term3 < 4000000) {
    $term1 = $term2;
    $term2 = $term3;
    $term3 = $term1 + $term2;

    if ($term3 % 2 == 0 && $term3 < 4000000) {
        $evenSum += $term3;
    }
}

print $evenSum . PHP_EOL;

File: solved/python/prob_2.py
"""
 * Wes Kilgour
 *
 * Problem 2
 * Even Fibonacci numbers
 *
 * https://projecteuler.net/problem=2
"""

fibTerms = [0, 1]

while fibTerms[-1] < 4000000:
    newTerm = fibTerms[-1] + fibTerms[-2]

    if newTerm > 4000000:
        break

    fibTerms.append(newTerm)

print(sum([i for i in fibTerms if i % 2 == 0]))
